market-contract/src/main.sw

```diff
@@ -857,112 +855,49 @@ fn match_order_internal(
-            match_price: match_price,
-        });
-        log_order_change_info(
+            b_order,
+            b_amount,
             q_id,
-            OrderChangeInfo::new(
-                OrderChangeType::OrderMatched,
-                block_height(),
-                matcher,
-                tx_id(),
-                q_order
-                    .amount,
-                q_order
-                    .amount - q_amount,
-            ),
+            q_order,
+            q_amount,
+            matcher,
+            match_price,
         );
-        log(MatchOrderEvent {
-            order_id: q_id,
-            asset: get_asset_id(AssetType::Quote),
-            order_matcher: matcher,
-            owner: q_order.owner,
-            counterparty: b_order.owner,
-            match_size: q_amount,
-            match_price: match_price,
-        });
-        log(TradeOrderEvent {
-            base_sell_order_id: b_id,
-            base_buy_order_id: q_id,
-            order_matcher: matcher,
-            trade_size: b_amount,
-            trade_price: match_price,
-            block_height: block_height(),
-            tx_id: tx_id(),
-        });
-        // update account balances
+
+        // Execute the trade and update balances
         execute_same_order_type_trade(b_order, q_order, b_amount, q_amount, price_delta);
-        // update order storages
-        if b_amount == b_order.amount {
-            matcher_reward += b_order.matcher_fee.as_u64();
-            remove_order(b_order.owner, b_id);
-        }
-        if q_amount == q_order.amount {
-            matcher_reward += q_order.matcher_fee.as_u64();
-            remove_order(q_order.owner, q_id);
-        }
-        if b_amount != b_order.amount {
-            let order_matcher_reward = b_order.matcher_fee.as_u64() * b_amount / b_order.amount;
-            matcher_reward += order_matcher_reward;
-            b_order.matcher_fee -= order_matcher_reward.try_as_u32().unwrap();
-            b_order.amount -= b_amount;
-            // update amount
-            storage.orders.insert(b_id, b_order);
-            return (MatchResult::PartialMatch, b_id, matcher_reward);
-        } else if q_amount != q_order.amount {
-            let order_matcher_reward = q_order.matcher_fee.as_u64() * q_amount / q_order.amount;
-            matcher_reward += order_matcher_reward;
-            q_order.matcher_fee -= order_matcher_reward.try_as_u32().unwrap();
-            q_order.amount -= q_amount;
-            // update amount
-            storage.orders.insert(q_id, q_order);
-            return (MatchResult::PartialMatch, q_id, matcher_reward);
-        }
-        (MatchResult::FullMatch, ZERO_B256, matcher_reward)
+        // Handle partial or full order fulfillment and rewards
+        update_order_storage_and_reward(
+            b_amount,
+            b_order,
+            b_id,
+            q_amount,
+            q_order,
+            q_id,
+            &mut matcher_reward,
+        )
```

Comment:
Refactored this to use a function `update_order_storage_and_reward` to remove nested / if else code blocks